lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -217,6 +217,23 @@ def visit_vcall(node)
         add_token(node.value.location, :method) unless special_method?(node.value.value)
       end
 
+      sig { params(node: SyntaxTree::ClassDeclaration).void }
+      def visit_class(node)
+        loc = node.constant.location
+        # let's treat the whole `Foo < Bar` span as a single class declaration token
+        # if there is a superclass.
```

Comment:
This means that the `<` will be included and highlighted differently than other punctuation. Could we emit two tokens for the class and superclass instead?

Reply:
This is on purpose, since I've noticed that `<` is already included in the current highlighting. If I split `constant` and `superclass`, then the `<` looks odd.

Reply:
It is punctuation **and** class:
![image](https://user-images.githubusercontent.com/224488/186686014-7c7dfc7d-00ff-4b34-ad18-b88f187fc1dc.png)
compared to:
![Screenshot 2022-08-25 at 17 03 04](https://user-images.githubusercontent.com/224488/186686161-9386b184-3acd-47a1-b201-7a1a14ecd381.png)

Reply:
Punctuation is higher up in the hierarchy, so the classification of the token is correct and we should not change it. Maybe the theme you're using doesn't define a colour for punctuation, which ends up inheriting from the wrong token type.